src/Network/Http/Auth/Oauth.php

```diff
@@ -135,6 +153,66 @@ protected function _hmacSha1($request, $credentials)
     }
 
     /**
+     * Use RSA-SHA1 signing.
+     *
+     * This method is suitable for plain HTTP or HTTPS.
+     *
+     * @param \Cake\Network\Http\Request $request The request object.
+     * @param array $credentials Authentication credentials.
+     * @return string
+     *
+     * @throws \RuntimeException
+     */
+    protected function _rsaSha1($request, $credentials)
+    {
+        if (!function_exists('openssl_pkey_get_private')) {
+            throw new \RuntimeException('RSA-SHA1 signature method requires the OpenSSL extension.');
+        }
+
+        $nonce = isset($credentials['nonce']) ? $credentials['nonce'] : Security::randomBytes(16);
+        $timestamp = isset($credentials['timestamp']) ? $credentials['timestamp'] : time();
+        $values = [
+            'oauth_version' => '1.0',
+            'oauth_nonce' => $nonce,
+            'oauth_timestamp' => $timestamp,
+            'oauth_signature_method' => 'RSA-SHA1',
+            'oauth_consumer_key' => $credentials['consumerKey'],
+        ];
+        if (isset($credentials['consumerSecret'])) {
+            $values['oauth_consumer_secret'] = $credentials['consumerSecret'];
+        }
+        if (isset($credentials['token'])) {
+            $values['oauth_token'] = $credentials['token'];
+        }
+        if (isset($credentials['tokenSecret'])) {
+            $values['oauth_token_secret'] = $credentials['tokenSecret'];
+        }
+        $baseString = $this->baseString($request, $values);
+
+        if (isset($credentials['realm'])) {
+            $values['oauth_realm'] = $credentials['realm'];
+        }
+
+        $credentials += [
+            'privateKeyPassphrase' => null,
+        ];
+        if (is_resource($credentials['privateKeyPassphrase'])) {
+            $resource = $credentials['privateKeyPassphrase'];
+            $passphrase = stream_get_line($resource, 0, PHP_EOL);
+            rewind($resource);
+            $credentials['privateKeyPassphrase'] = $passphrase;
+        }
+        $privateKey = openssl_pkey_get_private(file_get_contents($credentials['privateKeyFile']), $credentials['privateKeyPassphrase']);
```

Comment:
Do we want to change `privateKeyFile` to `privateKey` and to support `string` and `file` representations of the private key? One more opportunity for the developer to pass the private key.
And it would be consistent to the privateKeyPassphrase option (`string` and `file` representation).